src/components/InputRange.vue

```diff
@@ -28,19 +28,29 @@ const Input = require('gui/inputs/input');
 export default {
   mixins: [Input],
   data() {
-    const options = this.state.input.options.values[0];
-    const min = 1*options.min;
-    const max = 1*options.max;
-    const step = 1*options.Step;
+    const {min, max, Step:step} = this.state.input.options.values[0];
     return {
-      max,
       min,
-      step: step
+      max,
+      step
     }
   },
   methods: {
     checkValue() {
-      const valid = this.state.validate.required || !_.isEmpty(_.trim(this.state.value)) ? this.service.getValidator().validate(this.state.value) : true;
+      // if state required initial value is false
+      let valid = !this.state.validate.required;
+      // check if value of input is empty
+      const isEmpty = _.isEmpty(_.trim(this.state.value));
+      // in case of required input check if is not empty and check validity from validator
+      if (this.state.validate.required) {
+        valid =  !isEmpty && this.service.getValidator().validate(this.state.value);
+      } else if (isEmpty) {  //in case not required check if value is empty
+        //set default value
+        this.state.value = this.state.input.options.values[0].default;
+      } else {
+        // check validity from validator
+        valid = this.service.getValidator().validate(this.state.value);
+      }
       this.state.validate.valid = valid;
       this.change();
```

Comment:
Check if written like this below could be equivalent:

```suggestion
      // check if value of input is empty
      const isEmpty = _.isEmpty(_.trim(this.state.value));
      // set default value in case not required check if value is empty
      if (!this.state.validate.required && isEmpty) {
        this.state.value = this.state.input.options.values[0].default;
      }
      // TODO: add description
      this.state.validate.valid = this.state.validate.required && !isEmpty && this.service.getValidator().validate(this.state.value);
      this.change();
```

Reply:
@Raruto no. Because:
in case of this.state.validate.required === false has to be this.state.validate.valid = true if isEmpty, othewise  need to set valid valure according of this.service.getValidator().validate(this.state.value) response

Reply:
Ok, let's try to simplify it one step at a time, because there seems to me something strange or wrong in the overall logic (ref: that double call to the `this.service.getValidator().validate(this.state.value)` function).

So, currently, is it equivalent to the following?

```js
checkValue() {
  // check if value of input is empty
  const isEmpty = _.isEmpty(_.trim(this.state.value));

  // in case not required check if value is empty and set default value
  if (!this.state.validate.required && isEmpty) { 
    this.state.value = this.state.input.options.values[0].default;
  }

  // if state required initial value is false
  this.state.validate.valid = !this.state.validate.required;

  // if is not empty check validity from validator
  if (!isEmpty) {
    this.state.validate.valid = this.service.getValidator().validate(this.state.value);
  }

  this.change();
}
```

Reply:
@Raruto done